src/Security/Authentication.php

```diff
@@ -453,7 +453,7 @@ private function redirectForTwoFactorAuthentication(int $uid)
 	 */
 	public function setUnauthenticatedVisitor(string $url)
 	{
-		if (Strings::compareLink($this->session->get('visitor_home'), $url)) {
+		if (Strings::compareLink($this->session->get('visitor_home', ''), $url)) {
```

Comment:
Please use the short form of the ternary operator instead of the deprecated `$default` parameter of the `get()` method going forward.
```suggestion
		if (Strings::compareLink($this->session->get('visitor_home') ?: '', $url)) {
```
The `$default` parameter used to be interesting before the short form existed but it forces the evaluation of the default value even if it isn't necessary.